File: SortingAlogrithm/python/bubble sort.py
def bubble_sort(array):        #function to sort array using bubble sort
    swaps=0
    comparison=0
    for i in range(0,len(array)):
        for j in range(1,len(array)):
            if(array[j-1]>array[j]):
                swaps+=1
                temp=array[j]
                array[j]=array[j-1]
                array[j-1]=temp
            comparison+=1
        print("Array after ",i,"th pass is : ",array)
    print("Array after sorting is : ",array)
    print(swaps," number of swaps and",comparison," comparisons")

def input_array():                       #function for taking input of array
    attended=[]
    number_of_attendee=int(input("Enter total : "))
    for i in range(number_of_attendee):
        tep=int(input("Enter number :"))
        attended.append(tep)
    return attended

if __name__ == '__main__':
    arr=input_array()
    bubble_sort(arr)
    print("Array after sorting using selection sort : ",arr)
    

File: SortingAlgorithms/C++/Bubble_Sort.cpp
#include <iostream>
using namespace std;

void bubble_sort(int a[],int n){
	int t;
	for(int i=0;i<n-1;i++){
		for(int j=0;j<n-i-1;j++){
			if(a[j+1]<a[j]){
					t=a[j+1];
					a[j+1]=a[j];
					a[j]=t;
			}
		}
	}
}

 int main(){
	int n;
	cout<<"Enter the number of elements: ";
	cin>>n;
	int a[n];
	cout<<"Enter the elements: ";
	
	for(int i=0;i<n;i++){
		cin>>a[i];
	}

	bubble_sort(a,n);
	
	cout<<"Sorted array: ";
	for(int i=0;i<n;i++){
		cout<<a[i]<<" ";
	}
	cout<<endl;
	
	return 0;
}   